### PR TITLE
[juicefs] fix worker cache when set option

### LINKED ISSUE #2564: [BUG] Juice worker uses wrong cache when setting cache dir
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

When set cache-dir in worker options, juicefs worker will use different caches with fuse pod, and mount cache-dir in hostPath.

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by zwwhdls**
Signed-off-by: zwwhdls <zww@hdls.me>

### Ⅰ. Describe what this PR does
When set `cache-dir` in worker options, juicefs worker will use different caches with fuse pod, and mount cache-dir in hostPath.

### Ⅱ. Does this pull request fix one issue?
fixes #2564 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/juicefs/templates/fuse/daemonset.yaml [hunk 1]
```diff
@@ -137,15 +137,11 @@ spec:
               exec:
                 command: ["sh", "-c", "umount {{ .Values.fuse.mountPath }}"]
           volumeMounts:
-          - name: juicefs-fuse-mount
-            mountPath: {{ .Values.fuse.hostMountPath }}
-            mountPropagation: Bidirectional
-          - mountPath: /root/script
-            name: script
-          {{- range $name, $mount := .Values.cacheDirs }}
-          - name: cache-dir-{{ $name }}
```
Comment (cheyang) [inline charts/juicefs/templates/fuse/daemonset.yaml:146]: Please update changelog and chart version. thanks.

Comment (zwwhdls) [reply charts/juicefs/templates/fuse/daemonset.yaml:146]: done

**Pull request merged**